Number_types/include/CGAL/Quotient.h

```diff
@@ -686,11 +707,547 @@ template < class NT > class Real_embeddable_traits_quotient_base< Quotient<NT> >
+            x.den = -x.den;
+          } else if (x.num < 0 && x.den > 0) {
+            change_sign = true;
+            x.num = -x.num;
+          } else if (x.num > 0 && x.den < 0) {
+            change_sign = true;
+            x.den = -x.den;
+          }
+          CGAL_assertion(x.num > 0 && x.den > 0);
+
+          Protect_FPU_rounding<true> P(CGAL_FE_TONEAREST);
+          const auto n = get_interval_exp(x.num);
+          const auto d = get_interval_exp(x.den);
+
+          CGAL_assertion_msg(
+            CGAL::abs(double(n.second) - double(d.second)) < (1<<30)*2.0,
+            "Exponent overflow in Quotient<boost::multiprecision::cpp_int> to_interval()!");
+
+          const Interval_nt<> num(n.first);
+          const Interval_nt<> den(d.first);
+          const Interval_nt<> div = num / den;
+
+          const int e = static_cast<int>(n.second - d.second);
+          std::tie(l, u) = ldexp(div, e).pair();
+
+          if (change_sign) {
+            const double t = l;
+            l = -u;
+            u = -t;
+          }
+          return std::make_pair(l, u);
+        }
+
+        std::pair<double, double> get_interval_as_boost( Type& x ) const {
+
+          double l = 0.0, u = 0.0;
+          if (x.num == 0) { // return [0.0, 0.0]
+            return std::make_pair(l, u);
+          }
+          CGAL_assertion(x.num != 0);
+          CGAL_assertion(x.den != 0);
+
+          // Handle signs.
+          bool change_sign = false;
+          if (x.num < 0 && x.den < 0) {
+            x.num = -x.num;
+            x.den = -x.den;
+          } else if (x.num < 0 && x.den > 0) {
+            change_sign = true;
+            x.num = -x.num;
+          } else if (x.num > 0 && x.den < 0) {
+            change_sign = true;
+            x.den = -x.den;
+          }
+          CGAL_assertion(x.num > 0 && x.den > 0);
+
+          const int num_dbl_digits = std::numeric_limits<double>::digits;
+          const int shift = num_dbl_digits + (msb(x.den) - msb(x.num));
+          if (shift > 0) {
+            x.num <<= shift;
+          } else if (shift < 0) {
+            x.den <<= boost::multiprecision::detail::unsigned_abs(shift);
+          }
+          CGAL_assertion(int(msb(x.den) - msb(x.num)) == -53);
+
+          decltype(x.num) p, q, r;
+          boost::multiprecision::divide_qr(x.num, x.den, q, r);
+
+          if (r != 0) {
+            const int q_bits = msb(q);
+            if (q_bits == num_dbl_digits + 1) {
+              q <<= 1;
+              p = q;
+              ++q;
+            } else {
+              p = q;
+              ++q;
+            }
+          } else {
+            p = q;
+          }
+
+          l = boost::multiprecision::detail::do_cast<double>(p);
+          l = std::ldexp(l, -shift);
+          u = boost::multiprecision::detail::do_cast<double>(q);
+          u = std::ldexp(u, -shift);
+
+          if (change_sign) {
+            const double t = l;
+            l = -u;
+            u = -t;
+          }
```

Comment:
I added this
```

	  Type lb(l), ub(u);
	  assert(lb <= ub);
	  assert(lb <= x);
	  assert(ub >= x);
```   
and it asserts.

Reply:
I simplified the code/input, added these assertions, and now another assertion fails. See the [gist](https://gist.github.com/danston/44aa470c849565cb1e25f74d450ddaea) below.